intellij-ext/src/main/kotlin/com/metalbear/mirrord/MirrordEnabler.kt

```diff
@@ -35,10 +47,58 @@ class MirrordEnabler : ToggleAction() {
     override fun setSelected(e: AnActionEvent, state: Boolean) {
         if (state) {
             notify("mirrord enabled", NotificationType.INFORMATION, e.project)
+            e.project?.let { checkVersion(it) }
         } else {
             notify("mirrord disabled", NotificationType.INFORMATION, e.project)
         }
 
         MirrordListener.enabled = state
     }
+
+    private val pluginId = PluginId.getId("com.metalbear.mirrord")
+    private val version: String? = PluginManagerCore.getPlugin(pluginId)?.version
+    private val versionCheckEndpoint: String =
+        "https://version.mirrord.dev/get-latest-version?source=3&version=$version"
+    private var versionCheckDisabled
+        get() = PropertiesComponent.getInstance().getBoolean("versionCheckDisabled", false)
+        set(value) {
+            PropertiesComponent.getInstance().setValue("versionCheckDisabled", value)
+        }
+
+    private fun checkVersion(project: Project) {
+        val pc = PropertiesComponent.getInstance() // Don't pass project, to get ide-wide persistence.
+        val lastCheckEpoch = pc.getLong(LAST_CHECK_KEY, 0)
+        val nowUTC = LocalDateTime.now(ZoneOffset.UTC)
+        val lastCheckUTCDateTime = LocalDateTime.ofEpochSecond(lastCheckEpoch, 0, ZoneOffset.UTC)
+        if (lastCheckUTCDateTime.isAfter(nowUTC.minusHours(3))) {
+            return // Already checked in the last 3 hours. Don't check again yet.
+        }
+        val nowEpoch = nowUTC.toEpochSecond(ZoneOffset.UTC)
+        pc.setValue(LAST_CHECK_KEY, nowEpoch.toString())
+        val remoteVersion = Version.valueOf(URL(versionCheckEndpoint).readText())
+        val localVersion = Version.valueOf(version)
+        if (localVersion.lessThan(remoteVersion)) {
+            notifier(
+                "Your version of mirrord is outdated, you should update.",
```

Comment:
Maybe soften this phrase a little bit, something along the lines of:

```suggestion
                "Your version of mirrord is outdated. Do you want to update it now?",
```